addons/l10n_fr_sale_closure/models/account_closure.py

```diff
@@ -0,0 +1,171 @@
+
+        if not first_move and aml_aggregate['line_ids']:  # It's the first time we run for this frequency
+            aml_interval = self._refine_amls_for_interval(date_start, aml_aggregate['line_ids'])
+            total_interval = aml_interval.get('balance')
+        else:
+            total_beginning += previous_closure.total_beginning
+
+        # We keep the reference to avoid gaps (like daily object during the weekend)
+        last_move = first_move
+        if aml_aggregate['move_ids']:
+            last_move = last_move.browse(aml_aggregate['move_ids'][0])
+
+        return {'total_interval': total_interval,
+                'total_beginning': total_beginning,
+                'last_move_id': last_move.id,
+                'last_move_hash': last_move.l10n_fr_hash,
+                'date_closure_stop': interval_dates['date_stop'],
+                'date_closure_start': date_start}
+
+    def _interval_dates(self, frequency, company):
+        """
+        Method used to compute the theoretical date from which account move lines should be fetched
+        @param {string} frequency: a valid value of the selection field on the object (daily, monthly, annually)
+            frequencies are literal (daily means 24 hours and so on)
+        @param {recordset} company: the company for which the closure is done
+        @return {dict} the theoretical date from which account move lines are fetched.
+            date_stop date to which the move lines are fetched, always now()
+            the dates are in their Odoo Database string representation
+        """
+        date_stop = datetime.utcnow()
+        interval_from = None
+        if frequency == 'daily':
+            interval_from = date_stop - timedelta(days=1)
+        elif frequency == 'monthly':
+            month_target = date_stop.month > 1 and date_stop.month - 1 or 12
+            year_target = month_target < 12 and date_stop.year or date_stop.year - 1
+            interval_from = date_stop.replace(year=year_target, month=month_target)
+        elif frequency == 'annually':
+            year_target = date_stop.year - 1
+            interval_from = date_stop.replace(year=year_target)
+
+        return {'interval_from': FieldDateTime.to_string(interval_from),
+                'date_stop': FieldDateTime.to_string(date_stop)}
+
+    @api.multi
+    def write(self, vals):
+        raise UserError(WRITE_MSG)
+
+    @api.multi
+    def unlink(self):
+        raise UserError(WRITE_MSG)
+
+    @api.model
+    def automated_closure(self, frequency='daily'):
+        """To be executed by the CRON to create an object of the given frequency for each company that needs it
+        @param {string} frequency: a valid value of the selection field on the object (daily, monthly, annually)
+            frequencies are literal (daily means 24 hours and so on)
+        @return {recordset} all the objects created for the given frequency
+        """
+        def get_selection_value(field, value=''):
```

Comment:
If you change the value for the selection field from 'annually' to 'annual', you can probably get rid of this ugly function, and replace its calls with simply `_(value.capitalize())` (to be tested) ;-)